core/mount/mount_linux.go

```diff
@@ -266,9 +267,15 @@ func doPrepareIDMappedOverlay(tmpDir string, lowerDirs []string, usernsFd int) (
 		return nil, nil, err
 	}
 	cleanMount := func() {
-		if err := unix.Unmount(tempRemountsLocation, 0); err != nil {
-			log.L.WithError(err).Warnf("failed to unmount idmapped directory %s", tempRemountsLocation)
+		for retry := 0; retry < 5; retry++ {
```

Comment:
Its worth noting that we already have [this](https://github.com/containerd/containerd/blob/ef686423b1418a4840dbaef2696010c06f82d246/core/mount/mount_unix.go#L114) function that is getting us out of jail on the tmpmount overlayfs mount (it does retries).

Maybe we switch to that? Thanks for this change, I was going to propose it if I couldn't figure out the source of EBUSY proper, but since we're already doing this for the other mount here I think its appropriate to do now.

Reply:
Good point, thanks! I was sure there was a helper doing that, but I was on a rush and pushed the manual retry :-D

In fact, it's not that function but the one on top what we need: `Unmount()`. That one doesn't expect several mount points mounted on the same path.

Thanks!